### PR TITLE
[terra-clinical-onset-picker] Update docs to use mdx syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Updated docs to use MDX syntax.

### CODE REVIEW

File: packages/terra-clinical-onset-picker/README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The terra-clinical-onset-picker component provides users a way to enter or selec
 
 This component supports locales defined in the Terra component standards, except Arabic.
```
Comment (emilyrohrbough) [inline packages/terra-clinical-onset-picker/README.md]: @mjhenkes Should we log an issue to add Arabic so all terra-components meet the same translation standards?

File: packages/terra-clinical-onset-picker/src/terra-dev-site/doc/clinical-onset-picker/ClinicalOnsetPicker.1.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+** This component does not support Arabic translations
+
+[View component source code](https://github.com/cerner/terra-clinical/tree/master/packages/terra-clinical-onset-picker)
+
+## Examples
+<DefaultOnset title="Default with initialization" />
+<HandledOnset title="Onset picker with supplied data and handlers" />
+<OnsetWithHiddenLegend title="Onset picker with Hidden legend" />
+
+## Label Value View Props
```
Comment (emilyrohrbough) [inline packages/terra-clinical-onset-picker/src/terra-dev-site/doc/clinical-onset-picker/ClinicalOnsetPicker.1.doc.mdx]:
```suggestion
## Onset Picker Props
```

File: packages/terra-clinical-onset-picker/README.md (hunk 1 above)
Comment (mjhenkes) [reply packages/terra-clinical-onset-picker/README.md]: We don't actually support Arabic, (https://github.com/cerner/terra-aggregate-translations/blob/master/config/i18nSupportedLocales.js) even though we say we do (https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#internationalization-i18n-support)
My opinion is to remove the mentions to arabic from here and terra-ui.

Comment (mjhenkes) [reply packages/terra-clinical-onset-picker/README.md]:
```suggestion
This component supports locales defined in the Terra component standards.
```

Comment (emilyrohrbough) [reply packages/terra-clinical-onset-picker/README.md]: Ah right, we removed Arabic support a while back: https://github.com/cerner/terra-aggregate-translations/pull/2#issue-274424666

**Pull request merged**